### PR TITLE
fix: handle empty/uninitialized config/patch2 gracefully (#1811)

### LINKED ISSUE #1811: Patch Manager "Open Patch File" errors when config/patch2 is empty (show a "repository not initialized" message instead)
## Summary

When `config/patch2/` contains no patch files (the state a freshly‑downloaded standalone release ships in — five empty `FE6/FE7J/FE7U/FE8J/FE8U` stub folders), clicking **Open Patch File** in the Patch Manager throws an error instead of handling the empty/uninitialized state gracefully.

Reported in discussion #1808.

## Repro

1. Download & extract a standalone release (or empty `config/patch2/`).
2. Open a ROM → **Patch Manager**.
3. Click **Open Patch File**.
4. An error dialog appears; the patch list is empty.

## Root cause

`patch2` is no longer bundled in the release — it is delivered over Git (repo: `laqieer/FEBuilderGBA-patch2`, cloned into `config/patch2/`). A fresh install therefore has no patch payload, and the Patch Manager code path does not treat "patch repository not initialized" as an expected, recoverable state.

## Expected behaviour

- No error/exception when `config/patch2/` is empty or missing.
- Show a clear, actionable message, e.g. *"The patch database has not been downloaded yet. Use Initialize Repository / Check for Updates to fetch it."*
- Point the user at the initialize/update flow (see the feature issues linked below).

## Scope

**Ship to BOTH WinForms and Avalonia.** This is a blocking issue and necessary infrastructure work, so it is exempt from the usual "WinForms = bug‑fix only" policy — the empty‑repo state must be handled gracefully in both GUIs (and the CLI/Core path should not throw either).

## Related

- Discussion: #1808
- Workaround for users (manual patch2 setup): posted in #1808
- Feature work tracked separately:
  - #1812 — patch2 repository management (Initialize/Update + configurable URL)
  - #1813 — extend the same management to FE‑Repo / FE‑Repo‑Midi
  - #1814 — first‑run setup wizard for content repositories

Copilot CLI: 1.0.69-0
Model: Claude Opus 4.8 (claude-opus-4.8)

### PR DESCRIPTION
**Pull request by laqieer**
## Summary
On a fresh standalone release `config/patch2` ships as five empty `FE6/FE7J/FE7U/FE8J/FE8U` stub dirs (the patch library is git-delivered since #1766). In that state the WinForms Patch Manager's **Open Patch File** crashed: nothing is selected → `PatchFilename.Text==""` → `U.OpenURLOrFile("")` → `Process.Start("")` throws → `R.ShowStopError` dumped a **raw exception**. Both GUIs also showed a silent, unexplained empty patch list. Reported in discussion #1808.

## Changes (shipped to both GUIs + Core, infra-exempt per the issue)
**Core `PatchMetadataCore`**
- `NotInitializedMessage` — bilingual (JA+EN) not-downloaded-yet notice pointing at the in-app Check-for-Updates / Initialize flow (distinct from the Android-only `AndroidResourceNoticeCore.PatchLibraryUnavailableMessage`).
- `IsPatchLibraryEmpty(dir)` — true when the dir is missing/null or a **successful** scan finds no `PATCH_*.txt`; an existing dir that fails to enumerate returns **false** (a real error, not the fresh-install state) so callers don't mislabel it.

**WinForms** (bug-fix)
- `PatchOpenButton_Click`: guard empty `PatchFilename.Text` → show `NotInitializedMessage` when the library is empty, else a "select a patch first" hint; never call `Process.Start("")`.
- `ScanPatchs`: return empty for a **missing** patch dir instead of raising the generic `パッチ探索中にエラー` dialog on load/reload (adopted from the review board — the missing-dir path).

**Avalonia**
- Rename `ApplyAndroidEmptyStateNotice` → `ApplyEmptyStateNotice(patchDir)` and generalize: empty + `IsPatchLibraryEmpty(dir)` → desktop shows `NotInitializedMessage` (#1811), Android keeps `PatchLibraryUnavailableMessage` (#1641); populated clears the stale notice.

## Test plan
- [x] Core `IsPatchLibraryEmpty` (missing/null, empty dir, only-non-PATCH files, nested `PATCH_*.txt`) + `NotInitializedMessage` (non-empty, mentions download, distinct from the Android notice) — **5/5**.
- [x] Avalonia `AndroidResourceEmptyStateTests`: desktop-empty → `NotInitializedMessage`; Android-empty → `PatchLibraryUnavailableMessage`; populated → cleared — **5/5**.
- [x] Full suites regression-free: **Core 6456/0/6, Avalonia 9352/0/9, WinForms 1946/0**; WinForms x86 build 0 errors.
- [x] Verified the WinForms FE8 editor **loads cleanly with an empty `config/patch2`** (no raw scan-error dialog) — see `app-empty-patch2.png`.

## Screenshots
The tests directly assert the exact fixed behaviour (below). The friendly `NotInitializedMessage` itself is an edge-case dialog behind the first-run / patch2-download flows and a ROM-dependent view, which resisted reliable automated capture in this locked-down CI-agent environment (foreground-input lock + startup dialogs + UIA limitations); the behaviour is comprehensively covered by the 10 passing tests, and the app is shown loading cleanly in the empty-patch2 repro state.

#1811 tests passing
WinForms FE8 editor with empty patch2 loads without crashing

Closes #1811

Copilot CLI: 1.0.69-0
Model: Claude Opus 4.8 (claude-opus-4.8)

### CODE REVIEW

File: FEBuilderGBA.Avalonia/ViewModels/PatchManagerViewModel.cs [hunk 1]
```diff
@@ -156,33 +156,45 @@ public void LoadPatchList()
...
+            // empty list from an enumeration failure (files present but unreadable) must not be
+            // mislabelled as not-downloaded-yet.
+            if (!PatchMetadataCore.IsPatchLibraryEmpty(patchDir))
             {
-                // Idempotent: if the condition no longer holds (e.g. the VM is reused, or a future
-                // on-device delivery populates the list), clear our own stale notice so it cannot
-                // stick around. Only clears the Android notice — never clobbers an unrelated status.
-                StatusMessage = "";
+                return;
             }
```
[the remark this answers is not in the document]
Comment (laqieer) [reply FEBuilderGBA.Avalonia/ViewModels/PatchManagerViewModel.cs:193]: This early-return is intentional and in-scope-limited by design. When the list is empty but `IsPatchLibraryEmpty(patchDir)` is **false**, the patch directory exists and is readable-or-failed-to-enumerate — i.e. either a genuine filesystem error (permission / path-too-long) or a populated dir. We deliberately do **not** surface `NotInitializedMessage` there, because that would mislabel a real error (or a populated library) as "not downloaded yet" — the exact mis-messaging the cross-model review board asked us to avoid, and why `IsPatchLibraryEmpty` returns `false` (not `true`) on an enumeration exception.

Note `EnumeratePatches` already `Log.Error`s genuine enumeration failures, so those cases are not silent in the log. And because `ParsePatchFile` always returns a (possibly partial) entry per `PATCH_*.txt`, "files present but 0 parsed" cannot occur — a present `PATCH_*.txt` yields `Count > 0`. So the only path that reaches this early return is a real enumeration error, for which a distinct "failed to read patches" dialog is out of scope for #1811 (which targets the fresh-install *empty* patch2 state).

Copilot CLI: 1.0.69-0
Model: Claude Opus 4.8 (claude-opus-4.8)

**Comment by laqieer** [comment]
## Cross-Model Review Board

Branch B (Copilot CLI developer). Three independent reviewers.

| Reviewer | Verdict |
| --- | --- |
| `gemini-3.1-pro-preview` | **APPROVE** |
| `gpt-5.5` | APPROVE (after the `IsPatchLibraryEmpty` fix) |
| `claude-sonnet-5` | Independent build/test verification (no blocking issue surfaced) |

**Consensus: APPROVE.** No blocking code issues outstanding.

### Board finding addressed
- `gpt-5.5` (BLOCK): `IsPatchLibraryEmpty` used `!Directory.Exists → return true`, but `Directory.Exists` also returns `false` on an existence-probe error, so an *inaccessible* directory would be mislabelled "not initialized" — contradicting the documented contract. Fixed in `3c90e71ec`: drop the `Directory.Exists` gate; attempt `Directory.GetFiles` directly and distinguish `DirectoryNotFoundException` (genuinely missing → `true`) from other enumeration errors (real failure → `false`). Core `IsPatchLibraryEmpty` tests re-verified 5/5.
- `gemini-3.1-pro` confirmed the crash paths are closed (`Process.Start("")` guard + `ScanPatchs` missing-dir guard), the Avalonia `ApplyEmptyStateNotice` correctly surfaces `NotInitializedMessage` on desktop while preserving the #1641 Android notice, and the tests are isolated from the real submodule via temp paths.

### Bot thread
Resolved: the `ApplyEmptyStateNotice` early-return (empty list + `IsPatchLibraryEmpty==false`) is intentional — it avoids mislabelling a genuine enumeration error as "not downloaded yet" (the board's own guidance), and `EnumeratePatches` already logs real failures.

### Status
CI fully green; `MERGEABLE`/`CLEAN`; screenshots resolve from master (docs PR #1827 merged). Full suites: **Core 6456/0/6, Avalonia 9352/0/9, WinForms 1946/0**. Merging.

Copilot CLI: 1.0.69-0
Model: Claude Opus 4.8 (claude-opus-4.8)
